client/my-sites/email/hooks/use-password-reset-email-field.ts

```diff
@@ -0,0 +1,33 @@
+import { useSelector } from 'react-redux';
+import { HiddenFieldNames } from 'calypso/my-sites/email/form/mailboxes/components/new-mailbox-list/index';
+import { FIELD_PASSWORD_RESET_EMAIL } from 'calypso/my-sites/email/form/mailboxes/constants';
+import { getCurrentUserEmail } from 'calypso/state/current-user/selectors';
+
+type PasswordResetEmailFieldProps = {
+	selectedDomainName: string;
+	defaultHiddenFields?: HiddenFieldNames[];
+};
+
+type PasswordResetEmailFieldOptions = {
+	hiddenFields: HiddenFieldNames[];
+	passwordResetEmailFieldInitialValue: string;
+};
+
+export function usePasswordResetEmailField( {
+	selectedDomainName,
+	defaultHiddenFields = [],
+}: PasswordResetEmailFieldProps ): PasswordResetEmailFieldOptions {
+	const userEmail = useSelector( getCurrentUserEmail );
```

Comment:
Do we need to account for `userEmail` potentially being empty or null right after page load? I honestly don't know, but the thought struck me.

Reply:
I think we're okay as is. It could technically be empty/null, looking at the `getCurrentUserEmail` but I think that's more likely to be if this were ever called from a logged-out state. Looking at other implementations (including on this form already, where we end up printing the the email on screen while the field is hidden) I think we're in good shape.

If we wanted to be extra cautious though, we could check the truthiness of `useEmail` while pre-checking its validity, so the form would be displayed if the current user email was missing:

```
const isPasswordResetEmailValid =
	!! userEmail && ! new RegExp( `@${ selectedDomainName }$` ).test( userEmail );
```

Reply:
I tested by updating the email address of my user in wp-admin, and I never ran into a case when `userEmail` was empty to start with, so I think we're fine 👍